components/ml/org.wso2.carbon.ml.rest.api/src/main/java/org/wso2/carbon/ml/rest/api/ModelApiV10.java

```diff
@@ -187,6 +188,48 @@ public Response predict(@Multipart("modelId") long modelId, @Multipart("dataForm
         }
     }
 
+    /**
+     * Predict using a file and return predictions as a CSV.
+     */
+    @POST
+    @Path("/predictionStreams")
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    public Response streamingPredict(@Multipart("modelId") long modelId, @Multipart("dataFormat") String dataFormat,
+                            @Multipart("file") InputStream inputStream) {
+        PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+        int tenantId = carbonContext.getTenantId();
+        String userName = carbonContext.getUsername();
+        try {
+            // validate input parameters
+            // if it is a file upload, check whether the file is sent
+            if (inputStream == null || inputStream.available() == 0) {
+                String msg = String.format(
+                        "Error occurred while reading the file for model [id] %s of tenant [id] %s and [user] %s .", modelId,
+                        tenantId, userName);
+                logger.error(msg);
+                return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(msg).build();
+            }
+            final String predictions = mlModelHandler.streamingPredict(tenantId, userName, modelId, dataFormat, inputStream);
+            StreamingOutput stream = new StreamingOutput() {
+                @Override
+                public void write(OutputStream outputStream) throws IOException {
+                    Writer writer = new BufferedWriter(new OutputStreamWriter(outputStream));
+                    writer.write(predictions);
+                    writer.flush();
+                    writer.close();
+                }
+            };
+            return Response.ok(stream).header("Content-disposition", "attachment; filename=Predictions_" + modelId + "_" + MLUtils.getDate() + MLConstants.CSV).build();
+        } catch (Exception e) {
```

Comment:
Fixed in pull request https://github.com/wso2/carbon-ml/pull/51. Thanks for pointing this out and for the explanation!